src/ol_infrastructure/components/aws/olvpc.py

```diff
@@ -245,28 +248,54 @@ def __init__(self, vpc_config: OLVPCConfig, opts: Optional[ResourceOptions] = No
             )
             self.olvpc_subnets.append(ol_subnet)
         self.k8s_service_subnet = None
-        if vpc_config.k8s_service_subnet:
-            third_octet = (
+        self.k8s_pod_subnet = None
+        if vpc_config.k8s_service_subnet and vpc_config.k8s_pod_subnet:
+            service_subnet_third_octet = (
                 int(vpc_config.k8s_service_subnet.network_address) & 0xFF00
             ) >> 8
             self.k8s_service_subnet = ec2.Subnet(
-                f"{net_name}-k8s-service-subnet",
+                f"{vpc_config.vpc_name}-k8s-service-subnet",
                 cidr_block=str(vpc_config.k8s_service_subnet),
                 ipv6_cidr_block=self.olvpc.ipv6_cidr_block.apply(
-                    partial(subnet_v6, third_octet)
+                    partial(subnet_v6, service_subnet_third_octet)
                 ),
                 vpc_id=self.olvpc.id,
-                tags=vpc_config.merged_tags({"Name": net_name}),
+                tags=vpc_config.merged_tags(
+                    {"Name": f"{vpc_config.vpc_name}-k8s-service-subnet"}
+                ),
                 assign_ipv6_address_on_creation=True,
                 opts=resource_options,
             )
             ec2.RouteTableAssociation(
-                f"{net_name}-k8s-service-subnet-route-table-association",
+                f"{vpc_config.vpc_name}-k8s-service-subnet-route-table-association",
                 subnet_id=self.k8s_service_subnet.id,
                 route_table_id=self.route_table.id,
                 opts=resource_options,
             )
 
+            pod_subnet_third_octet = (
+                int(vpc_config.k8s_pod_subnet.network_address) & 0xFF00
```

Comment:
Can you add a comment detailing the purpose of this? It is rather opaque at first glance and some commentary around it will help with future maintenance.